lib/OpenQA/WebAPI/Controller/API/V1/Iso.pm

```diff
@@ -77,7 +77,15 @@ sub _parse_dep_variable {
 
     my @after = split(/\s*,\s*/, $value);
 
-    return map { $_ . ':' . $settings->{MACHINE} } @after;
+    return map {
+        if ($_ !~ /^(.+):([^:]+)$/) {
+            [$_, $settings->{MACHINE}];
+        }
+        else {
+            my ($test, $machine) = split /:/, $_;
+            [$test, $machine];
```

Comment:
We might check other users of `_parse_dep_variable()` getting now array ref instead of string.

Reply:
fixed